software/controller/lib/debug/debug.h

```diff
@@ -17,116 +17,122 @@ limitations under the License.
 #define DEBUG_H
 
 #include "circular_buffer.h"
+#include "trace.h"
 #include <optional>
 #include <stdint.h>
 
+namespace Debug {
+
 // States for the internal state machine
-enum class DbgPollState { WAIT_CMD, PROCESS_CMD, SEND_RESP };
+enum class State { kWait, kProcessing, kResponding };
 
 // The binary serial interface uses two special characters
 // These values are pretty arbitrary.
 // See debug.cpp for a detailed description of how this works
-enum class DbgSpecial : uint8_t { ESC = 0xf1, TERM = 0xf2 };
-
-enum class DbgCmdCode : uint8_t {
-  MODE = 0x00,            // Return the current firmware mode
-  PEEK = 0x01,            // Peek into RAM
-  POKE = 0x02,            // Poke values into RAM
-  PRINT_BUFF_READ = 0x03, // Read strings from the print buffer
-  VAR = 0x04,             // Variable access
-  TRACE = 0x05,           // Data trace commands
+enum class SpecialChar : uint8_t { kEscape = 0xf1, kEndTransfer = 0xf2 };
+
+enum class ErrorCode : uint8_t {
+  kNone = 0x00,            // No error (=success)
+  kCrcError = 0x01,        // CRC error on command
+  kUnknownCommand = 0x02,  // Unknown command code received
+  kMissingData = 0x03,     // Not enough data passed with command
+  kNoMemory = 0x04,        // Insufficient memory
+  kInternalError = 0x05,   // Some type of interal error (aka bug)
+  kUnknownVariable = 0x06, // The requested variable ID is invalid
+  kInvalidData = 0x07,     // data is out of range
 };
 
-enum class DbgErrCode : uint8_t {
-  OK = 0x00,           // success
-  CRC_ERR = 0x01,      // CRC error on command
-  BAD_CMD = 0x02,      // Unknown command code received
-  MISSING_DATA = 0x03, // Not enough data passed with command
-  NO_MEMORY = 0x04,    // Insufficient memory
-  INTERNAL = 0x05,     // Some type of interal error (aka bug)
-  BAD_VARID = 0x06,    // The requested variable ID is invalid
-  RANGE = 0x07,        // data out of range
+namespace Command {
+
+enum class Code : uint8_t {
+  kMode = 0x00,     // Return the current firmware mode
+  kPeek = 0x01,     // Peek into RAM
+  kPoke = 0x02,     // Poke values into RAM
+  kConsole = 0x03,  // Read strings from the print buffer
+  kVariable = 0x04, // Variable access
+  kTrace = 0x05,    // Data trace commands
 };
 
-struct CmdContext {
-  const uint8_t *req;
-  const int req_len;
-  uint8_t *resp;
-  const int max_resp_len;
-  int resp_len;
+// Structure that represents a command's parameters
+struct Context {
+  const uint8_t *request;             // pointer to the request (command's data)
+  const uint32_t request_length;      // length of the request
```

Comment:
@martukas Since this is a "protocol" I think it makes sense to be explicit about the size. `size_t` will be different when unit testing on x86 which could lead to different results.

Reply:
Fair enough